### PR TITLE
Fixes installer for PHP 8.1 compatibility

### PR DESCRIPTION
**Pull request by SilinMykola**

### CODE REVIEW

**Review by andrewbess — APPROVED** [review]

File: tests/MagentoHackathon/Composer/Magento/FullStack/AbstractTest.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public static function setUpBeforeClass()
         @unlink(self::getBasePath().'/magento-modules/vendor/theseer/directoryscanner/tests/_data/nested/empty');
         
         $process = new Process(
-            self::getComposerCommand().' archive --format=zip --dir="tests/FullStackTest/artifact" -vvv',
+            [self::getComposerCommand().' archive --format=zip --dir="tests/FullStackTest/artifact" -vvv'],
```
Comment (xmav) [inline tests/MagentoHackathon/Composer/Magento/FullStack/AbstractTest.php:41]: With this change we need to set minimal version for symfony/process in composer json as v2.x expect type string as argument

**Review by xmav — CHANGES_REQUESTED** [review]

File: tests/MagentoHackathon/Composer/Magento/FullStack/AbstractTest.php (hunk 1 above)
Comment (SilinMykola) [reply tests/MagentoHackathon/Composer/Magento/FullStack/AbstractTest.php:41]: I changed minimal version for symfony/process in composer json for version where attribute in construct method has new type (string|array).

Comment (SilinMykola) [reply tests/MagentoHackathon/Composer/Magento/FullStack/AbstractTest.php:41]: @xmav Our constraint allow to install  any versions. I checked with minimum versions for me. Composer installed the symfony/process v5.3.7. I think we can leave as is.

File: src/MagentoHackathon/Composer/Magento/PackageXmlParser.php [hunk 2]
```diff
@@ -23,16 +23,17 @@ class PackageXmlParser extends PathTranslationParser
...
-    protected $_targets = array();
+    protected $_targets = [];
 
     /**
      * Constructor
      *
      * @param string $moduleDir
      * @param string $packageXmlFile
-     * @param array  $translations
+     * @param string $pathSuffix
```
[the remark this answers is not in the document]
Comment (SilinMykola) [reply src/MagentoHackathon/Composer/Magento/PackageXmlParser.php:33]: @fascinosum Thanks! I fixed.

File: tests/MagentoHackathon/Composer/Magento/InstallerTest.php [hunk 3]
```diff
@@ -125,113 +125,113 @@ public function testGetParser( $packageExtra, $expectedClass, $composerExtra, $p
...
-            array(
+        return [
+            [
                 'method' => 'copy',
                 'expectedClass' => 'MagentoHackathon\Composer\Magento\Deploystrategy\Copy',
-                'composerExtra' => array(  ),
                 'packageName'   => 'example/test1',
-            ),
-            array(
+                'composerExtra' => [],
```
[the remark this answers is not in the document]
Comment (SilinMykola) [reply tests/MagentoHackathon/Composer/Magento/InstallerTest.php:138]: @fascinosum Thanks! I fixed.

**Pull request merged**